coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,640 @@
+                                       system_matrix,
+                                       solution,
+                                       system_rhs);
+  }
+  {
+    // Afterwards, we apply the coupling boundary condition. The `boundary_data`
+    // has already been filled by preCICE.
+    MatrixTools::apply_boundary_values(boundary_data,
+                                       system_matrix,
+                                       solution,
+                                       system_rhs);
+  }
+}
+
+
+
+template <int dim>
+void
+CoupledLaplaceProblem<dim>::solve()
+{
+  SolverControl            solver_control(1000, 1e-12);
+  SolverCG<Vector<double>> solver(solver_control);
+  solver.solve(system_matrix, solution, system_rhs, PreconditionIdentity());
+
+  std::cout << "   " << solver_control.last_step()
+            << " CG iterations needed to obtain convergence." << std::endl;
+}
+
+
+
+template <int dim>
+void
+CoupledLaplaceProblem<dim>::output_results() const
+{
+  DataOut<dim> data_out;
+
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+
+  data_out.build_patches();
+
+  std::ofstream output("solution-" + std::to_string(time_step) + ".vtk");
+  data_out.write_vtk(output);
+}
+
+
+
+template <int dim>
+void
+CoupledLaplaceProblem<dim>::run()
+{
+  std::cout << "Solving problem in " << dim << " space dimensions."
+            << std::endl;
+
+  make_grid();
+  setup_system();
+
+  // After we set up out system, we initialize preCICE using the functionalities
+  // of our Adapter.
+  delta_t = adapter.initialize(dof_handler, boundary_data);
+
+  // preCICE steers the coupled simulation completely: `isCouplingOngoing` is
+  // used to synchronize the end of the simulation with the coupling partner
+  while (adapter.precice.isCouplingOngoing())
+    {
+      // The time step number is solely used to generate unique output files
+      ++time_step;
+      // In the time loop, we assemble the coupled system and solve it, as
+      // usual. According to our configuration, we obtained already the data of
+      // the `fany_boundary_condition` participant during the initialization.
+      assemble_system();
+      solve();
+
+      // After we solved the system, we advance with our system to the next time
+      // level. In a coupled simulation, we would pass our calculated data to
+      // preCICE and obtain data from other participants. Here, we simply obtain
+      // data from the other participant.
+      delta_t = adapter.advance(boundary_data, delta_t);
+
+      // In case our time step has been completed, we write the results to an
+      // output file.
+      if (adapter.precice.isTimeWindowComplete())
+        output_results();
+    }
+}
+
+
+
+int
+main(int argc, char **argv)
+{
+  // TODO: Should we keep the MPI dependency here? It makes things more clear in
+  // the initialization of the Adapter, but requires DEAL_II_WITH_MPI to be
+  // turned on I guess.
```

Comment:
No, let's remove it, I would say.

Reply:
Done, 
I renamed the respective variables (commRank..) consistent with deal.II, not with the dummy.